libgadget/winds.c

```diff
@@ -252,8 +255,17 @@ int cmp_by_part_id(const void * a, const void * b)
 
 /* Find the 1D DM velocity dispersion of the winds by running a density loop.*/
 static void
-winds_find_weights(TreeWalk * tw, struct WindPriv * priv, int * NewStars, int NumNewStars, const double Time, const double hubble, ForceTree * tree)
+winds_find_weights(TreeWalk * tw, struct WindPriv * priv, int * NewStars, int NumNewStars, const double Time, const double hubble, ForceTree * tree, DomainDecomp * ddecomp)
 {
+    /* Flags that we need to free the tree to preserve memory order*/
+    priv->tree_alloc_in_wind = 0;
```

Comment:
The logic here reads like:

1. If there is a tree, use it
2. If there is no tree, build one, use it, then free it.

The usual way of implementing such logic is checking tree == NULL; So checking tree->tree_allocated_flag looks a bit surprising. I guess it is complicated to change the calling convention everywhere.

Reply:
Indeed - it is not the tree that should be == NULL, but the tree->Nodes pointer and that has its own oddities with being offset from Nodes_base.